src/mtcp/mtcp_restart.c

```diff
@@ -773,10 +775,12 @@ main(int argc, char *argv[], char **environ)
       //         mtcp_restart is statically linked, and doesn't need it.
       Area heap_area;
       MTCP_ASSERT(getMappedArea(&heap_area, "[heap]") == 1);
-      start1 = max(heap_area.endAddr, lh_info.memRange.end);
+      start1 = (char *) max((uint64_t) heap_area.endAddr,
+                            (uint64_t) lh_info.memRange.end);
       Area stack_area;
       MTCP_ASSERT(getMappedArea(&stack_area, "[stack]") == 1);
-      end1 = min(stack_area.endAddr - 4 * GB, highMemStart - 4 * GB);
+      end1 = (char *) min((uint64_t) stack_area.endAddr - 4 * GB,
+                          (uint64_t) highMemStart - 4 * GB);
```

Comment:
We should be very careful when performing `minus` on `uint64_t`, making sure the first is greater than the second.